### ISSUE TITLE
Add allOf/anyOf/oneOf concept

### ISSUE THREAD

**Issue by n0rdlicht** [bhch/django-jsonform#39]
One of the very useful features in JSON Schemas is the option to have multiple options via [`allOf/anyOf/oneOf`](https://json-schema.org/draft/2020-12/json-schema-core.html#logic) for the same key/set of keys when nesting. For our project github.com/cividi/spatial-data-package-platform this would solve a major hurdle for fully embracing django-jsonform as an elegant solution for tenant defined forms. Would be awesome to know if this is at all thinkable as a future addition. Thanks so much for your efforts.

**Comment by bhch** [comment]
Yeah, currently I've implemented a custom, non-standard json schema specification. But I'm working towards adding the standard features. I've got some features lined up for the next release (like referencing and recursive nesting). I'll try to include `allOf/anyOf/oneOf` also. I'll have to see.

Currently, for basic string based options, you can use the [`choices` keyword](https://django-jsonform.readthedocs.io/en/stable/guide/choices.html). And for `anyOf` type options, there's a [`multiselect` widget](https://django-jsonform.readthedocs.io/en/stable/guide/choices.html#multiple-selections).

**Comment by bhch** [comment]
I'm starting work on this. But these keywords - `allOf/anyOf/oneOf` - are mainly for validation and I'm having a hard time trying to figure out how to render a form for them.

If you can give me a couple of example schemas for them and describe the form's expected behaviour, it will make things a bit easier for me. Thanks.

**Comment by bhch** [comment]
@n0rdlicht Hi, I'm still waiting for suggestions from you. Please, see my previous comment. Thank you.

**Comment by danihodovic** [comment]
oneOf is useful as it allows us to map Django TextChoices cleanly to a select field. Otherwise one must use the pre_save_hook to transform the choice label to the choice value.

**Comment by danihodovic** [comment]
@bhch Many thanks for this excellent library. It's probably the best tool I've found for unstructured data that needs a visual representation. I'm using it for field translations, but I can think of a dozen other use-cases for early stage prototyping. This project deserves more attention.

**Comment by gersmann** [comment]
A great use case for this would be to build forms for variable field sets for inheritance-simiar discriminated unions, as they are for instance implemented by Pydantic. 
https://docs.pydantic.dev/usage/types/#discriminated-unions-aka-tagged-unions

This is a very common JSON/JSON Schema pattern.

**Comment by bhch** [comment]
Hello, everyone participating in this thread!

I've just implemented this feature on the JS side. You can test it out in the playground here: https://bhch.github.io/react-json-form/playground/#18-anyof.

I encourage you to kindly test with  your own schemas because there might be certain cases which I've forgot to account for. If you do find any bugs, please report. But any sort of feedback is appreciated.

I'll release the Django app next weekend.

Thank you.

**Comment by gersmann** [comment]
Worked like a charm on my test schema. Very nice, thats a game changer for working with Pydantic in Django.

**Comment by bhch** [comment]
This feature has been released in [v2.16.0](https://github.com/bhch/django-jsonform/releases/tag/v2.16.0). Please update and test it out.

There are a few limitations in their implementation, so do [read the docs](https://django-jsonform.readthedocs.io/en/stable/schema.html#oneof-anyof-allof) as well.

Thank you.
